### PR TITLE
Gather coverage data for unit tests and upload to codecov

### PR DESCRIPTION
**Pull request by rprimet**

### CODE REVIEW

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -22,12 +22,22 @@ jobs:
...
+          ./ci/install_mpi.sh
         env:
           MPI_INSTALL: ${{ matrix.mpi_install }}
-      - run: pip install pytest pytest-env
-      - run: pip install -e .
-      - run: echo "localhost slots=16">hostfile
-      - run: pytest
+      - name: Run unit tests
+        run: |
+          pip install pytest pytest-env
```
Comment (tomMoral) [inline .github/workflows/unittests.yml:34]: Why not put everything in `setup.cfg`?

**Review by tomMoral — COMMENTED** [review]
A small comment but it looks good overall!

File: .github/workflows/unittests.yml (hunk 1 above)
Comment (rprimet) [reply .github/workflows/unittests.yml]: Good point, this should be fixed now

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -22,12 +22,21 @@ jobs:
...
+        run: |
+          pip install -e .[test]
+          echo "localhost slots=16">hostfile
+          pytest --cov-report=xml --cov=dicodile
+      - name: Upload coverage
+        uses: codecov/codecov-action@v1
+        with:
+          flags: unittests 
+          fail_ci_if_error: true 
+          verbose: true
```
Comment (tomMoral) [inline .github/workflows/unittests.yml:42]: Missingn an extra line?
```suggestion
          verbose: true

```

**Review by tomMoral — APPROVED** [review]
LGTM!

File: .github/workflows/unittests.yml (hunk 2 above)
Comment (rprimet) [reply .github/workflows/unittests.yml]: Fixed, thanks!

**Comment by tomMoral** [comment]
It seems that we still don't see CI run in PR.
I guess we have a config issue but I don't know why.

Could you check this?

Also, it looks like there is an issue in the config as we cannot see the coverage report in files: https://codecov.io/gh/rprimet/dicodile/src/97d67379b8a6f65bf1d3de6adfcc1ac3344261f3/dicodile/update_z/coordinate_descent.py

**Comment by rprimet** [comment]
> It seems that we still don't see CI run in PR.
> I guess we have a config issue but I don't know why.
> 
> Could you check this?

Yes, there's a config issue related to the action triggers I guess. We could revert it to `[push, pull_requests]` ?

> 
> Also, it looks like there is an issue in the config as we cannot see the coverage report in files: https://codecov.io/gh/rprimet/dicodile/src/97d67379b8a6f65bf1d3de6adfcc1ac3344261f3/dicodile/update_z/coordinate_descent.py

I'm having no issue seeing the coverage report at this link, could you confirm that it doesn't work for you?

**Comment by tomMoral** [comment]
> Yes, there's a config issue related to the action triggers I guess. We could revert it to `[push, pull_requests]` ?

Perfect Thanks! It was indeed the setup as the default branch in this repo is `master` and not `main` but I guess using all pushes is fine.

> I'm having no issue seeing the coverage report at this link, could you confirm that it doesn't work for you?

It does work now, I don't know why it failed when I checked but it is fine now.
I will merge this one! Thanks

**Pull request merged**